q2_feature_classifier/_skl.py

```diff
@@ -6,11 +6,55 @@
 # The full license is in the file LICENSE, distributed with this software.
 # ----------------------------------------------------------------------------
 
+from dataclasses import dataclass, field
+from functools import cached_property
 from itertools import islice, repeat
-from copy import deepcopy
+from typing import Dict, List
 
 from joblib import Parallel, delayed
 
+
+@dataclass
+class _TaxonNode:
+    # The _TaxonNode is used to build a hierarchy from a list of sorted class
+    # labels. It allows one to quickly find class label indices of taxonomy
+    # labels that satisfy a given taxonomy hierarchy. For example, given the
+    # 'k__Bacteria' taxon, the _TaxonNode.range property will yield all class
+    # label indices where 'd__Bacteria' is a prefix.
```

Comment:
Sorry @SirDavidLudwig, did you mean k__Bacteria here?

Reply:
Sorry about that. Just pushed the fix.